sm/plugins/tedge_docker_plugin/tedge_docker_plugin.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/sh
+
+set -e
+
+COMMAND="$1"
+IMAGE_NAME="$2"
+IMAGE_TAG="$3"
+
+case "$COMMAND" in
+    prepare)
+        # nothing to do here
+        ;;
+    list)
+        sudo docker image list --format '{"name":"{{.Repository}}","version":"{{.Tag}}"}'
+        ;;
+    install)
+        # Stop all containers using the provided image name
+        sudo docker stop $(docker ps --format "{{.ID}} {{.Image}}" | grep $IMAGE_NAME | awk '{print $1}')
```

Comment:
I didn't use the `docker ps --filter ancestor=$IMAGE_NAME` option to filter out the containers using the given image name because it will not just show the containers using the given image **directly** but all the containers that may be using this image **at any of its ancestral layers**.